src/main/java/generic/PagesDictionary.java

```diff
@@ -0,0 +1,28 @@
+package generic;
+
+import com.google.common.collect.ImmutableList;
+import common.AbstractPage;
+import pages.Bing;
+import pages.Google;
+
+import java.util.List;
+import java.util.Optional;
+
+public final class PagesDictionary {
+
+    private static final List<PageBinding> DICTIONARY = ImmutableList.of(
```

Comment:
It seems to me here you brake Open Closed Principle because to add any new Page need to make changes here